src/components/OrderBookBtn.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react'
+import styled from 'styled-components'
+import { DEFAULT_MODAL_OPTIONS, ModalBodyWrapper } from 'components/Modal'
+import Modali, { useModali } from 'modali'
+import OrderBookWidget from './OrderBookWidget'
+import { TokenDetails } from 'types'
+import { useWalletConnection } from 'hooks/useWalletConnection'
+import { safeTokenName } from '@gnosis.pm/dex-js'
+import TokenSelector from './TokenSelector'
+import useSafeState from 'hooks/useSafeState'
+import { useTokenList } from 'hooks/useTokenList'
+
+const ModalWrapper = styled(ModalBodyWrapper)``
+
+interface OrderBookBtnProps {
+  baseToken: TokenDetails
+  quoteToken: TokenDetails
+  label?: string
+  className?: string
+}
+
+export const OrderBookBtn: React.FC<OrderBookBtnProps> = (props: OrderBookBtnProps) => {
+  const { baseToken: baseTokenDefault, quoteToken: quoteTokenDefault, label, className } = props
+  const { networkIdOrDefault: networkId } = useWalletConnection()
+  const tokenList = useTokenList(networkId)
+  const [baseToken, setBaseToken] = useSafeState<TokenDetails>(baseTokenDefault)
+  const [quoteToken, setQuoteToken] = useSafeState<TokenDetails>(quoteTokenDefault)
+
+  const [modalHook, toggleModal] = useModali({
+    ...DEFAULT_MODAL_OPTIONS,
+    large: true,
+    title: `${safeTokenName(baseToken)}-${safeTokenName(quoteToken)} Order book`,
+    message: (
+      <ModalWrapper>
+        <TokenSelector tokens={tokenList} selected={baseToken} onChange={setBaseToken} />
+        <TokenSelector tokens={tokenList} selected={quoteToken} onChange={setQuoteToken} />
+        <OrderBookWidget baseToken={baseToken} quoteToken={quoteToken} networkId={networkId} />
+      </ModalWrapper>
+    ),
+    buttons: [<Modali.Button label="Close" key="yes" isStyleDefault onClick={(): void => modalHook.hide()} />],
```

Comment:
Why are you using `modalHook.hide()` instead of `toggleModal`? Not that it matters as they are the same function. But we use toggle everywhere else.

Reply:
I just saw the hide function in their API and I thought it was more declarative than the toggle for the close button and accept button

U don't like it? :)

Reply:
I honestly don't care. Same function, even if named hide, also toggles
:+1:

Reply:
> I honestly don't care. Same function, even if named hide, also toggles.

what?? hahaha. what a weird library
I'll use toggle from now on